clang/include/clang/Interpreter/CodeCompletion.h

```diff
@@ -0,0 +1,53 @@
+//===--- InterpreterAutoCompletion.h - Incremental Utils --------*- C++ -*-===//
+//
+// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+
+#ifndef LLVM_CLANG_INTERPRETER_CODE_COMPLETION_H
+#define LLVM_CLANG_INTERPRETER_CODE_COMPLETION_H
+#include "clang/Sema/CodeCompleteConsumer.h"
+#include "Interpreter.h"
+#include "llvm/LineEditor/LineEditor.h"
```

Comment:
I find it a little surprising to see the line editor included here. Why do we need it?

Reply:
because on Line 47 we have  `std::vector<llvm::LineEditor::Completion> operator()(llvm::StringRef Buffer, size_t Pos) const;`

Reply:
I understand but I am pretty sure we can convert if from `std::vector<std::string>` to `std::vector<llvm::LineEditor::Completion>`.

Reply:
@vgvassilev My understanding is we can't.  The documentation `setListCompleter()` says:

> Set the completer for this [LineEditor](https://llvm.org/doxygen/classllvm_1_1LineEditor.html) to the given list completer.
A list completer is a function object which takes arguments of type [StringRef](https://llvm.org/doxygen/classllvm_1_1StringRef.html) (the string to complete) and size_t (the zero-based cursor position in the [StringRef](https://llvm.org/doxygen/classllvm_1_1StringRef.html)) and returns a std::vector<Completion>.

and this is how completions are obtained:

```
  template <typename T>
  struct ListCompleterModel : ListCompleterConcept {
    ListCompleterModel(T Value) : Value(std::move(Value)) {}
    std::vector<Completion> getCompletions(StringRef Buffer,
                                           size_t Pos) const override {
      return Value(Buffer, Pos);
    }
    T Value;
  };
```
https://llvm.org/doxygen/LineEditor_8h_source.html#l00139

A `ReplListCompleter`'s operator() method is invoked in the way above. My sense is that the method needs to
1. be public
2. return `std::vector<Completion>`

Reply:
We need to figure out how to not expose it in the Interpreter.h API, I think.